File: PresentationTier/PresentationTier/Model/Product.cs
namespace PresentationTier.Model
{
    public class Product
    {
        public string ProductId { get; set; }  // Unique identifier for the product
        public string ProductName { get; set; } // Name of the product
        public decimal Price { get; set; } // Price of the product

        public Product()
        {
            ProductId = "";
            ProductName = "";
            Price = 0;
        }
        public Product(string Id, string name, decimal price)
        {
            this.ProductId = Id;
            ProductName = name;
            Price = price;
        }
    }
}

File: T2_ApplicationTier_SpringBoot/src/main/java/org/example/t2_applicationtier_springboot/model/Product.java
package org.example.t2_applicationtier_springboot.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Product {
    @JsonProperty("ProductId")
    private String productId;
    @JsonProperty("ProductName")
    private String productName;
    @JsonProperty("Price")
    private double price;

    // Constructors
    public Product() {}

    public Product(String productId, String productName, double price) {
        this.productId = productId;
        this.productName = productName;
        this.price = price;
    }

    // Getters and Setters
    public String getProductId() {
        return productId;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }
}
